pkg/serviceprovider/oauthcapability.go

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceprovider
+
+import (
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/serviceprovider/oauth"
+)
+
+type OAuthCapability interface {
+	// GetOAuthEndpoint returns the URL of the OAuth initiation. This must point to the SPI oauth service, NOT
+	//the service provider itself.
+	GetOAuthEndpoint() string
```

Comment:
Weren't we thinking about deducing this from the SP name somwhere in the shared code, so that we can get rid of this method? 

IMHO, it would be a good time to do it now when we're extracting this new interface.

Reply:
it is in one place for all service providers that want to use oauth https://github.com/redhat-appstudio/service-provider-integration-operator/pull/400/files#diff-6516ccdb48901b1538af9dbffac762657beecb3b975490d6c5cec7395302da9cR38